### DISCUSSION TITLE
About calling C + + Libraries

### DISCUSSION THREAD

**Discussion by XinPeiHou** [milvus-io/milvus#4159] [Q&A and General discussion] [answered]
The grpc service I wrote needs to call the C + + library of Milvus
Milvus version：0.10.1

The version of libprotoc on my computer is 3.13.0
The compilation passed, but a segment error was prompted during execution
As fellows：
:~/facesearchcloudservice/build/bin$ ./facesearchcloudserivce
Segmentation fault (core dumped)
houxinpei@hjimi-k8s-node02:~/facesearchcloudservice/build/bin$ gdb ./facesearchcloudserivce
GNU gdb (Ubuntu 8.1-0ubuntu3) 8.1.0.20180409-git
Copyright (C) 2018 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.  Type "show copying"
and "show warranty" for details.
This GDB was configured as "x86_64-linux-gnu".
Type "show configuration" for configuration details.
For bug reporting instructions, please see:
<http://www.gnu.org/software/gdb/bugs/>.
Find the GDB manual and other documentation resources online at:
<http://www.gnu.org/software/gdb/documentation/>.
For help, type "help".
Type "apropos word" to search for commands related to "word"...
Reading symbols from ./facesearchcloudserivce...done.
(gdb) r
Starting program: /home/houxinpei/facesearchcloudservice/build/bin/facesearchcloudserivce
[Thread debugging using libthread_db enabled]
Using host libthread_db library "/lib/x86_64-linux-gnu/libthread_db.so.1".

Program received signal SIGSEGV, Segmentation fault.
0x00007ffff79eea13 in google::protobuf::internal::AddDescriptors (table=0x7ffff7543fc0 <descriptor_table_google_2fprotobuf_2fany_2eproto>) at google/protobuf/generated_message_reflection.cc:2412
2412    google/protobuf/generated_message_reflection.cc: No such file or directory.
(gdb) where
#0  0x00007ffff79eea13 in google::protobuf::internal::AddDescriptors (table=0x7ffff7543fc0 <descriptor_table_google_2fprotobuf_2fany_2eproto>) at google/protobuf/generated_message_reflection.cc:2412
#1  0x00007ffff7de5783 in call_init (env=0x7fffffffe2e8, argv=0x7fffffffe2d8, argc=1, l=<optimized out>) at dl-init.c:72
#2  _dl_init (main_map=0x7ffff7ffe170, argc=1, argv=0x7fffffffe2d8, env=0x7fffffffe2e8) at dl-init.c:119
#3  0x00007ffff7dd60ca in _dl_start_user () from /lib64/ld-linux-x86-64.so.2
#4  0x0000000000000001 in ?? ()
#5  0x00007fffffffe577 in ?? ()
#6  0x0000000000000000 in ?? ()
(gdb)

**Comment by fishpenguin** [comment]
Can you run the sdk_simple in your machine?

**Comment by fishpenguin** [comment, marked as answer]
You'd better not keep multiple version protobuffer in your machine. This bug is most likely caused by the multiple version  of protobuffer.
